osi_object.proto

```diff
@@ -745,66 +748,133 @@ message MovingObject
             //
             // Definition: Hatchback car with maximum length 4 m.
             //
+            // This is one of the OSI types which map from "car" in both OpenScenario 1.x and 2.x.
+            //
             TYPE_SMALL_CAR = 2;
 
             // Vehicle is a compact car.
             //
             // Definition: Hatchback car with length between 4 and 4.5 m.
             //
+            // This is one of the OSI types which map from "car" in both OpenScenario 1.x and 2.x.
+            //
             TYPE_COMPACT_CAR = 3;
 
             // Vehicle is a medium car.
             //
             // Definition: Hatchback or sedan with lenght between 4.5 and 5 m.
             //
+            // This is one of the OSI types which map from "car" in both OpenScenario 1.x and 2.x.
+            //
             TYPE_MEDIUM_CAR = 4;
 
             // Vehicle is a luxury  car.
             //
             // Definition: Sedan or coupe that is longer then 5 m.
             //
+            // This is one of the OSI types which map from "car" in both OpenScenario 1.x and 2.x.
+            //
             TYPE_LUXURY_CAR = 5;
 
             // Vehicle is a delivery van.
             //
             // Definition: A delivery van.
             //
+            // This corresponds to "van" in OpenScenario 1.x.
+            //
+            // This is one of the OSI types which map from "truck" in OpenScenario 2.x.
+            //
             TYPE_DELIVERY_VAN = 6;
 
-            // Vehicle is a heavy truck.
+            // Vehicle is a (heavy) truck.
+            //
+            // Definition: A rigid-body truck that cannot pull a semi-trailer.
+            //
+            // This corresponds to "truck" in OpenScenario 1.x.
+            //
+            // This is one of the OSI types which map from "truck" in OpenScenario 2.x.
             //
             TYPE_HEAVY_TRUCK = 7;
 
-            // Vehicle is a truck with semitrailer.
+            // Vehicle is a tractor unit capable of pulling a semi-trailer.
+            //
+            // Definition: The cab for an articulated vehicle.
+            //
+            // This has no equivalent in OpenScenario 1.x.
+            //
+            // This is one of the OSI types which map from "truck" in OpenScenario 2.x.
+            //
+            TYPE_SEMITRACTOR = 16;
+
+            // This vehicle is a trailer unit for an articulated vehicle.
```

Comment:
Suggestion:
This vehicle is a semi-trailer that can be pulled by a semi-tractor.

Definition: A semi-trailer is a trailer without a front axle. The front part of the semi-trailer is mounted on the semi-tractor which leads to a significant overlap of the bounding-boxes of semi-tractor and semi-trailer.

Reply:
I'd have thought most trailers don't have a front axle?  A quick google search of "car trailer" suggests most have only rear or mid axles.

Perhaps the distinction is instead weight or size rather than axles or joining mechanism / overlap?

Reply:
There are different trailer types, e.g., trailers with one axle in the middle or a tandem trailer. These are mainly towed by vehicles or agricultural vehicles.
Trailers towed by trucks mostly have either a front and a rear axle (often referred to as “full-trailer”) or just a rear-axle (“semi-trailer”). The semi-trailer is towed by a semi-tractor, which is not able to carry additional loads, but its only purpose is to tow the semi-trailer. “full-trailer” can be towed by trucks or agricultural vehicles. One specialty is a dolly, which is an adapter to couple semi-trailer to trucks or agricultural vehicle or generate road trains consisting of semi-trailers. This dolly variant transforms semi-trailers into full-trailers and should be handled like full-trailers.

I think semi-trailers should be handled with a separate type since they are widely used and have special characteristics due to their coupling mechanism. In that regard, it also makes to have semi-tractors as separate types to describe the pulling part for semi-trailers.

It might be reasonable to introduce the full-trailer as an additional trailer variant. By this we would have:
full-trailer: for big trailers with front and rear axle
Semi-trailer: for semi-tractor + semi-trailer couplings
Trailer: describing the general vehicle trailers

Would that be an approach?

Reply:
As discussed in the harmonisation meeting.  This is a complex area and trying to define precise boundaries between types is more than we want to take on as part of this PR which is primarily about clarifying semi-trailer / tractor definitions.

As such, I have remove any new attempts at defining boundaries, and I have written a note in the top level enum comment indicating that the imprecise boundaries are deliberate and left to the implementor to decide on in their specific use case.

Does that work for you?